scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/Scheduler.java

```diff
@@ -924,4 +925,59 @@ String getTaskServerLogsByTag(String id, String taskTag) throws UnknownJobExcept
      * These are meaningful users for accounting {@link SchedulerUsage}
      */
     List<SchedulerUserInfo> getUsersWithJobs() throws NotConnectedException, PermissionException;
+    
+    
+    
+    /**
+     * Retrieve a tasks names list from the scheduler.
+     * 
+     * @param sessionId  a valid session id
+     * @param taskTag  a complete tag to use to filter tasks
+     * @param from  the starting date to fetch tasks from. The format is in Epoch time.
+     * @param to  the end date to stop fetching tasks. The format is in Epoch time.
+     * @param mytasks  <code>True</code> will only fetch the user tasks, <code>False</code> will fetch everyones.
+     * @param running  fetch the running tasks.
+     * @param pending  fetch the pending tasks.
+     * @param finished  fetch the finished tasks.
+     * @param offset  the starting task to include in the paginated list.
+     * @param limit  the last task (not included) before stopping fetching tasks in the paginated list.
+     * @return  the paginated list of tasks names satisfying the given criterias. The total number of tasks (without pagination() is also returned.
+     * @throws NotConnectedException
+     * @throws PermissionException
+     */
+    Page<TaskId> getTaskIds(String taskTag, long from, long to, boolean mytasks, boolean running,
+            boolean pending, boolean finished, int offset, int limit)
+                    throws NotConnectedException, PermissionException;
+
+    /**
+     * Retrieve a taskstates list from the scheduler.
+     * 
+     * @param sessionId  a valid session id
+     * @param taskTag  a complete tag to use to filter tasks
+     * @param from  the starting date to fetch tasks from. The format is in Epoch time.
+     * @param to  the end date to stop fetching tasks. The format is in Epoch time.
+     * @param mytasks  <code>True</code> will only fetch the user tasks, <code>False</code> will fetch everyones.
+     * @param running  fetch the running tasks.
+     * @param pending  fetch the pending tasks.
+     * @param finished  fetch the finished tasks.
+     * @param offset  the starting task to include in the paginated list.
+     * @param limit  the last task (not included) before stopping fetching tasks in the paginated list.
+     * @return  the paginated list of taskstates satisfying the given criterias. The total number of tasks (without pagination() is also returned.
+     * @throws NotConnectedException
+     * @throws PermissionException
+     */
+    Page<TaskState> getTaskStates(String taskTag, long from, long to, boolean mytasks,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: org.ow2.proactive.scheduler.common.exception.NotConnectedException, org.ow2.proactive.scheduler.common.exception.PermissionException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1160)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 9 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)